### PR TITLE
add image to user profile

### PR DESCRIPTION
**Pull request by smanini1**
#### Description:

Added UIImagePicker to select image from library
Squared image to fit circle imageView
Save image with updateUserProfile
Set image when loading UserProfile

---

#### Preview:

ezgif com-resize

### CODE REVIEW

File: ios-target/Extensions/UIImageViewExtension.swift [hunk 1]
```diff
@@ -13,4 +13,30 @@ extension UIImageView {
   func setImage(with url: URL?, placeholder: String) {
     sd_setImage(with: url, placeholderImage: UIImage(named: placeholder), options: .refreshCached)
   }
+  
+  func setRoundedShape(borderColor: UIColor, borderWidth: CGFloat) {
+    layer.borderWidth = 1
```
Comment (germanStabile) [inline ios-target/Extensions/UIImageViewExtension.swift:18]: use the borderWidth parameter here

File: ios-target/Extensions/UIImageViewExtension.swift [hunk 2]
```diff
@@ -13,4 +13,30 @@ extension UIImageView {
   func setImage(with url: URL?, placeholder: String) {
     sd_setImage(with: url, placeholderImage: UIImage(named: placeholder), options: .refreshCached)
   }
+  
+  func setRoundedShape(borderColor: UIColor, borderWidth: CGFloat) {
```
Comment (germanStabile) [inline ios-target/Extensions/UIImageViewExtension.swift]: you can default to 1 here

File: ios-target/Extensions/UIImageViewExtension.swift [hunk 3]
```diff
@@ -13,4 +13,30 @@ extension UIImageView {
...
+    clipsToBounds = true
+  }
+  
+  func cropsToSquare() {
+    guard
+      let originalImage = image?.cgImage,
+      let imageOrientation = image?.imageOrientation
+    else { return }
+    let imageWidth = CGFloat((originalImage.width))
+    let imageHeight = CGFloat((originalImage.height))
```
Comment (germanStabile) [inline ios-target/Extensions/UIImageViewExtension.swift]: why double parenthesis ?

File: ios-target/Extensions/UIImageViewExtension.swift [hunk 4]
```diff
@@ -13,4 +13,30 @@ extension UIImageView {
...
+    let cropSize = imageWidth > imageHeight ? imageHeight : imageWidth
```
Comment (germanStabile) [inline ios-target/Extensions/UIImageViewExtension.swift]: `let cropSize = max(imageHeight, imageWidth)`

File: ios-target/Extensions/UIImageViewExtension.swift [hunk 5]
```diff
@@ -13,4 +13,30 @@ extension UIImageView {
...
+  
+  func setRoundedShape(borderColor: UIColor, borderWidth: CGFloat) {
+    layer.borderWidth = 1
+    layer.masksToBounds = false
+    layer.borderColor = borderColor.cgColor
+    layer.cornerRadius = frame.height / 2
+    clipsToBounds = true
+  }
+  
+  func cropsToSquare() {
```
Comment (germanStabile) [inline ios-target/Extensions/UIImageViewExtension.swift]: maybe 'cropToSquare'?

**Review by germanStabile — APPROVED** [review]
LG!

File: ios-target/Extensions/UIImageViewExtension.swift (hunk 1 above)
Comment (smanini1) [reply ios-target/Extensions/UIImageViewExtension.swift:18]: ups!

File: ios-target/Extensions/UIImageViewExtension.swift (hunk 4 above)
Comment (germanStabile) [reply ios-target/Extensions/UIImageViewExtension.swift]: sorry should be min

**Review by mcousillas6 — APPROVED** [review]

**Pull request merged**